### PR TITLE
Enable Code Coverage in Live Tests

### LINKED ISSUE #33150: [FEATURE REQ] Setup Code Coverage for Java Live Tests
**Is your feature request related to a problem? Please describe.**
I was working on Service reliability program for Identity SDK and as a part of that we need to ensure we have code coverage captured across all the platforms. Currently, code coverage is not setup for Live Tests across all packages inside communication folder.
[Pipelines - Runs for java - azure-communication-identity - tests](https://dev.azure.com/azure-sdk/internal/_build?definitionId=2729) 

**Describe the solution you'd like**
Not sure about the exact solution, but this is my findings :

`Steps "Generate pom for aggregate code coverage report", "Generate aggregate code coverage report" and "PublishCodeCoverageResults" are disabled in the pipeline. `

`These steps are enabled based on the condition : `
`_condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['JavaTestVersion'], '1.17'), eq('${{ parameters.SDKType }}', 'client'), eq('${{ parameters.SkipAggregateReports }}', 'false'))_ `

`Values for params "SDKType" and "SkipAggregateReports" are not setup correctly in "live.tests.yml" due to which the condition are failing. `

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Fixes #33150 

Enables capturing code coverage in the Live tests pipelines.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by benbp — APPROVED** [review]

**Review by JimSuplizio — APPROVED** [review]
@alzimmermsft, I'm going to approve this, but I do have a question. Do we want to turn this on all the time, or do we want to make this a parameter that individual services can opt into for their pipelines?

**Comment by alzimmermsft** [comment]
> @alzimmermsft, I'm going to approve this, but I do have a question. Do we want to turn this on all the time, or do we want to make this a parameter that individual services can opt into for their pipelines?

The change in the PR makes it appear that test coverage will run all the time but coverage is only gathered in one job based on this logic:

`condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['JavaTestVersion'], '1.17'), eq('${{ parameters.SDKType }}', 'client'), eq('${{ parameters.SkipAggregateReports }}', 'false'))`

https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/steps/build-and-test.yml#L76

@benbp and I were discussing that this should follow what is done for the FromSource runs where this is something explicitly passed by the test matrix but that may happen later

**Review by srnagar — APPROVED** [review]

File: eng/pipelines/templates/steps/build-and-test.yml [hunk 1]
```diff
@@ -59,7 +56,10 @@ steps:
     displayName: 'Run tests'
     inputs:
       mavenPomFile: pom.xml
-      options: '$(DefaultTestOptions) -DAZURE_TEST_DEBUG=$(IsDebug) -Dparallel-test-playback=${{ parameters.ParallelTestPlayback }} ${{ parameters.TestOptions }} -T ${{ parameters.TestParallelization }}'
+      ${{ if eq(variables['RunAggregateReports'], 'true') }}:
```
Comment (benbp) [inline eng/pipelines/templates/steps/build-and-test.yml:59]: This will not work, at the time it's evaluated the variable will always be undefined.

Comment (benbp) [reply eng/pipelines/templates/steps/build-and-test.yml:59]: You may need a prior powershell step that processes all the options and sets them into a final variable that gets consumed by the maven task. Arguably this is easier to maintain than doing all the yaml global stuff anyway at this point.

Comment (alzimmermsft) [reply eng/pipelines/templates/steps/build-and-test.yml]: This has been completely reverted as I've been misreading `javadoc.skip` as `jacoco.skip`.

**Pull request merged**